The project TestExceptionHelper demonstrates the usage of the class ExceptionHelper.
This helps to handle properly the exception TargetInvocationException,
with proper rethrowing the InnerException, at the same time preserving the call stack.

For more details see the code of MainForm.cs, and the code of ExceptionHelper itself.